tests/queryEditor.spec.ts

```diff
@@ -135,3 +135,35 @@ test('"Timeseries custom query" multiple ts OK', async ({ selectors, readProvisi
   const tsWithUnits = tsExternalIds.map((ts, i) => `${ts}-${units[i]}`);
   await expect(panelEditPage.panel.fieldNames).toContainText(tsWithUnits);
 });
+
+
+test('"Event query" as table is OK', async ({ page, gotoDashboardPage, readProvisionedDashboard, grafanaVersion }) => {
+  const dashboard = await readProvisionedDashboard({ fileName: 'weather-station.json' });
+  const dashboardPage = await gotoDashboardPage(dashboard);
+  
+  const panelEditPage = await dashboardPage.gotoPanelEditPage('3')
+
+  await expect(panelEditPage.panel.fieldNames).toContainText(["externalId", "description", "startTime", "endTime"]);
+
+  const query = `
+    {
+      "prefix": {
+          "property": ["externalId"],
+          "value": "test_event (1"
+      }
+    }`;
+
+  if (semver.gte(grafanaVersion, '10.2.0')) {
+    await page.getByTestId(/Code editor container/).getByRole("textbox").first().fill(query, { force: true });
+  } else {
+    await page.getByLabel(/Code editor container/).getByRole("textbox").first().fill(query, { force: true });
+  }
+
+  await waitForQueriesToFinish(page, grafanaVersion);
+  await expect(panelEditPage.refreshPanel({ waitForResponsePredicateCallback: isCdfResponse('/events/list') })).toBeOK();
+
+  await expect.poll(async () => {
+    const cellTexts = await panelEditPage.panel.data.getByRole('cell').allInnerTexts();
+    return cellTexts.every(text => /test_event \(0/.test(text)) && !cellTexts.some(text => /test_event \(1-9/.test(text));
```

Comment:
maybe we can make this more readable?

Suggestion below (Im not 100% sure if this is what you meant to write as well):

````
const EVENT_PREFIX_PATTERN = /test_event \(0/;
const EXCLUDED_EVENT_PATTERN = /test_event \(1-9/;

const validateCellTexts = (cellTexts: string[]) => {
  const hasValidPrefix = cellTexts.every(text => EVENT_PREFIX_PATTERN.test(text));
  const hasNoExcludedEvents = !cellTexts.some(text => EXCLUDED_EVENT_PATTERN.test(text));
  return hasValidPrefix && hasNoExcludedEvents;
};

Reply:
yes, used you code now. thanks